internal/pkg/cli/task_run.go

```diff
@@ -945,6 +981,77 @@ func (o *runTaskOpts) deploy() error {
 	return o.deployer.DeployTask(os.Stderr, input, deployOpts...)
 }
 
+// deployEnvFileIfNeeded uploads the env file if needed, ensures that an S3 bucket is available, and returns the ARN of uploaded file.
+func (o *runTaskOpts) deployEnvFileIfNeeded() (string, error) {
+	if o.envFile == "" {
+		return "", nil
+	}
+
+	if o.envFileARN != "" {
+		return o.envFileARN, nil
+	}
+
+	region := aws.StringValue(o.sess.Config.Region)
+
+	app, err := o.store.GetApplication(o.appName)
+	if err != nil {
+		return "", fmt.Errorf("get application from store: %w", err)
+	}
+
+	// bootstrap S3 bucket
+	o.spinner.Start(fmt.Sprintf(fmtTaskRunResourcesStart, color.HighlightUserInput(o.appName)))
+	err = o.deployer.AddPipelineResourcesToApp(app, region)
+	if err != nil {
+		o.spinner.Stop(log.Serrorf(fmtTaskRunResourcesFailed, color.HighlightUserInput(o.appName)))
+		return "", fmt.Errorf("add env file bucket to application %s in %s: %w", o.appName, region, err)
+	}
+	o.spinner.Stop(log.Ssuccessf(fmtTaskRunResourcesComplete, color.HighlightUserInput(o.appName)))
+
+	// push env file
+	o.spinner.Start(fmt.Sprintf(fmtTaskRunEnvUploadStart, color.HighlightUserInput(o.envFile)))
+	envFileARN, err := o.pushEnvFileToS3(app, region)
+	if err != nil {
+		o.spinner.Stop(log.Serrorf(fmtTaskRunEnvUploadFailed, color.HighlightUserInput(o.envFile)))
+		return "", err
+	}
+	o.spinner.Stop(log.Ssuccessf(fmtTaskRunEnvUploadComplete, color.HighlightUserInput(o.envFile)))
+
+	o.envFileARN = envFileARN
```

Comment:
nit: is there any benefit in caching it? it doesn't seem like this method will be called multiple times

Reply:
This actually is called multiple times! When a built image is provided, it first deploys the task stack to create the repo and everything, then builds the image & pushes it to the ECR repo, and then deploys the task stack again. Since this function is called from deploy(), it'll be called twice in that case otherwise.